Program.cs

```diff
@@ -76,58 +136,66 @@ private static List<string> ProcessArguments(string[] args)
 			return filesToSearch;
 		}
 
-		static bool Check(string password, string filename)
+		static bool Check(string asHex, string filename)
         {
-            string ByteArrayToString(byte[] ba)
+            var buffer = new byte[SHA1LENGTH];
+            using var sr = File.OpenRead(filename);
+            
+            //Number of lines
+            //var high = (sr.Length / (LINELENGTH + 2)) - 1;
+            //As line length is now variable just use the bytes in the file:
+            long high = sr.Length;
+            long low = 0L;
+
+            int ch;
+
+            while (low <= high)
             {
-                string hex = BitConverter.ToString(ba);
-                return hex.Replace("-", "");
-            }
-
-            var bytes = System.Text.Encoding.UTF8.GetBytes(password);
-            var sha = System.Security.Cryptography.SHA1.Create();
-            var passwordBytes = sha.ComputeHash(bytes);
-            var asHex = ByteArrayToString(passwordBytes);
+                long middle = (low + high + 1) / 2;
+                //Console.WriteLine($"{low}:{middle}:{high}");
+                
+                //sr.Seek((LINELENGTH + 2) * ((long)middle), SeekOrigin.Begin);
+
+				//seek to next cr/newline (0d0a) ahead of current position
+                bool do_seek = true;
+				while (middle >= low+1 && do_seek)
+				{
+					middle -= 1;
+					sr.Seek(middle, SeekOrigin.Begin);
+					ch = sr.ReadByte();
+					if (ch == 0x0d)
```

Comment:
Can you try changing the if statement to be

```
	if (ch == 0x0a)
	{
		// Linux / Mac line feeds
		if (middle < sr.Length - 1)
		{
			middle += 1;
			do_seek = false;
		}
	}
	else if (ch == 0x0d)
	{
		// Windows line feeds
		ch = sr.ReadByte();
		if (ch == 0x0a && middle < sr.Length - 1)
		{
			middle += 2;
			do_seek = false;
		}
	}

```

It works on my Mac,  but I assume you are using a Windows machine. 


Assuming it works on your machine - I'll merge it in - thanks :-)

Reply:
I can test for all platforms. For the download of HaveIBeenPawnd it comes with 0x0d 0x0a windows new lines independent of the clients OS. My guess is, people don't do a `dos2unix` on a file this size. On the other side, there are other files to use as input. I think there is no hurry, so I give it some more thoughts ... 

Above code does not work as intended, as it only finds the 0x0d if it as by chance already on 0x0a. In nearly all times it runs into the Linux/Mac branch. Perhaps just looking for 0x0a works on all OS? I will dive into it again next week ...